patientslikeme/drugsDict.py

```diff
@@ -0,0 +1,76 @@
+from bs4 import BeautifulSoup
+import requests
+import urllib
+
+def initializeSoup(url):
+    request = requests.get(url)
+    soup = BeautifulSoup(request.text, 'html.parser')
+    return soup
+
+def getDrugName(url):
+    soup = initializeSoup(url)
+    #removes js and css
+    for script in soup (["script", "style"]):
+        script.extract()
+    return soup.h1.span.string
+    
+def addBrandNames(url,dictionary):
+    soup = initializeSoup(url)
+    brandNames = soup.find("span", {"class":"comma-separated"})
+    if(brandNames is not None):
+        brandNames = brandNames.find_all('a')
+        #print(brandNames)
+        for brand in brandNames:
+            url = brand.get("href")
+            fullUrl = "https://www.patientslikeme.com" + url
+            name = brand.string.rstrip().encode('utf-8')
+            dictionary[name] = fullUrl
+
+#this will be our drug dicitonary
+drugDict = {}
+website = "https://www.patientslikeme.com/treatments/browse?cat="
+
+#creating dictionary for drugs
+drugDict = {}
+allCat = [1,2]
+for category in allCat:
+    soup = initializeSoup(website + str(category).encode('utf-8'))
+    #button for next page
+    disabled = soup.find("a", {"class":"button icon-only is-not-rwd next_page",
+                               "disabled":"true"})
+    #checks to see if there is only one page
+    if(disabled is not None):
+        drugTable = soup.find("table", {"id":"tbl-treatments"})
+        drugs = drugTable.findChildren('tr')
+        #iterate through each drug on the page
+        for i in range(1, len(drugs)):
+                for drugUrl in drugs[i].findChildren("a"):
+                    url = drugUrl.get("href")
+                    fullUrl = "https://www.patientslikeme.com" + url
+                    # go to URL for each drug, add each name and associating url to the dictionary
+                    drugDict[getDrugName(fullUrl)] = fullUrl
+    else:
+        #loop iterates through each page until it reaches the last page
+        pageNum = 1
+        while(disabled is None):
+            #print(pageNum)
+            # get URLs of all drugs
+            website = "https://www.patientslikeme.com/treatments/browse?cat=" + str(category).encode('utf-8') + "&page=" + str(pageNum).encode('utf-8')
+            soup = initializeSoup(website)
+            drugTable = soup.find("table", {"id":"tbl-treatments"})
+            drugs = drugTable.findChildren('tr')
+            disabled = soup.find("a", {"class":"button icon-only is-not-rwd next_page",
+                                   "disabled":"true"})    
+            #iterate through each drug on the page
+            for i in range(1, len(drugs)):
+                for drugUrl in drugs[i].findChildren("a"):
```

Comment:
Looks like the same code block as in line 47. Try to abstract the logic to another function?